### ISSUE TITLE
AdGuard can't create a VPN connection after last Samsung S10+ update

### ISSUE THREAD

**Issue by roaidl** [AdguardTeam/AdguardForAndroid#3529]
### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [ o ] I am running the latest version
- [ o ] I checked the documentation and found no answer
- [ o ] I checked to make sure that this issue has not already been filed

### Issue Details

- **AdGuard version:**
  -  3.5 nightly8
- **Filtering mode:**
  -  Local VPN
- **Device:**
  -   galaxy-s10+
- **Operating system and version:**
  -  Android 10
- **Root access:**
  -  no 

### Expected Behavior

AdGuard kan ikke skabe en VPN forbindelse. Det er muligt, at VPN ikke er understøttet af din firmware.

### Actual Behavior
All test on https://adguard.com/ko/test.html  failed. Ads leaking.

### Screenshots

<details><summary>Screenshot:</summary>

</details>

### Additional Information
Adguard cannot setup VPN connection after last Samsung S10+ update 08/2020
Issue #3521 Even if I did the same, it failed. 

plz... help me

**Comment by admitrevskiy** [comment]
Hi! The latest Samsung update is a bit of headache for us :)

Please confirm that you've tried both options:
- Dual Messanger disabling
- Granting permission with ADB

If so, could you attach a screenshot with an attempt to grant a permission?

**Comment by roaidl** [comment]
There was no problem when using dual messenger before the firmware update, but is there a problem now?

And when I try ADB for the first time, NOX breaks or what?

**Comment by admitrevskiy** [comment]
> There was no problem when using dual messenger before the firmware update, but is there a problem now?

Yep, there were no problems with Dual Messenger before update, but new Dual Messenger somehow attaches Restricted Accounts API. I have no idea why Samsung engineers decided to use it, but it's so. Here is an explanation [how it works](https://github.com/AdguardTeam/AdguardForAndroid/issues/3299#issuecomment-667160839) and our [article at the Knowledge Base](https://kb.adguard.com/en/android/solving-problems/restricted-profile)

> And when I try ADB for the first time, NOX breaks or what?

Since Dual Messenger uses Restricted Profile features, any app that creates a VPN and tries to exclude some packages from it needs `com.adguard.android android.permission.INTERACT_ACROSS_USERS`. This is a system-level permission and there are only 3 ways to workaround this issue
- Get rid of restricted account (the simplest one)
- Grant a permission via ADB (there is no way to deal with it via UI)
- Use Local HTTP Proxy mode (for rooted devices)

In short: 
- ADB does not break anything. It's just a way to interact with the device at the system level.
- Permission granting should not affect anything except for permission

**Comment by roaidl** [comment]
Interact across users Through permission, Adguard solved.
Other apps can be resolved by distributing the app by adding the permission to the manifest. (Share of app maker)

adb shell pm grant com.adguard.android android.permission.INTERACT_ACROSS_USERS 
Can I do it like this?

**Comment by admitrevskiy** [comment]
> adb shell pm grant com.adguard.android android.permission.INTERACT_ACROSS_USERS
> Can I do it like this?

Yes, is should solve the problem
